gpt_index/readers/download.py

```diff
@@ -0,0 +1,67 @@
+"""Download loader from the Loader Hub."""
+
+import json
+import os
+import subprocess
+import sys
+from importlib import util
+
+from langchain.utilities import RequestsWrapper
+
+from gpt_index.readers.base import BaseReader
+
+LOADER_HUB_URL = (
+    "https://raw.githubusercontent.com/emptycrown/loader-hub/main/loader_hub"
+)
+
+
+def download_loader(loaderClassName: str) -> BaseReader:
+    """Download a single loader from the Loader Hub.
+
+    Args:
+        loaderClassName: The name of the loader class you want to download,
+            such as `SimpleWebPageReader`.
+    Returns:
+        A Loader.
+    """
+    requests = RequestsWrapper()
+    response = requests.run(f"{LOADER_HUB_URL}/library.json")
+    library = json.loads(response)
+
+    # Look up the loader id (e.g. `web/simple_web`)
+    loader_id = library[loaderClassName]["id"]
+    dirpath = ".modules"
+    loader_filename = loader_id.replace("/", "-")
```

Comment:
ooc why do this? why not preserve the existing directory structure?

Reply:
Felt easier to not have to recursively create directories -- don't feel strongly about this though